Subject: The gnu java compiler cannot compile this class
Author: Simon Frankenberger <simon-alpine@fraho.eu>
Upstream: No

1. ERROR in /builds/bratkartoffel/aports/community/openjdk7/src/icedtea-2.6.23/openjdk-boot/jdk/src/share/classes/sun/security/util/Optional.java (at line 196)
    return Optional.ofNullable(mapper.apply(value));
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Type mismatch: cannot convert from Optional<capture#5-of ? extends U> to Optional<U>


--- openjdk.orig/jdk/src/share/classes/sun/security/util/Optional.java
+++ openjdk/jdk/src/share/classes/sun/security/util/Optional.java
@@ -193,7 +193,7 @@
         if (!isPresent())
             return empty();
         else {
-            return Optional.ofNullable(mapper.apply(value));
+            return (Optional) Optional.ofNullable(mapper.apply(value));
         }
     }
 
